crates/bevy_math/src/bounding/raytest3d.rs

```diff
@@ -105,6 +105,80 @@ impl IntersectsVolume<BoundingSphere> for RayTest3d {
     }
 }
 
+/// An intersection test that casts an [`Aabb3d`] along a ray.
+#[derive(Debug)]
+pub struct AabbCast3d {
+    /// The ray along which to cast the bounding volume
+    pub ray: RayTest3d,
+    /// The aabb that is being cast
+    pub aabb: Aabb3d,
+}
+
+impl AabbCast3d {
+    /// Construct an [`AabbCast3d`] from an [`Aabb3d`], origin, [`Direction3d`] and max distance.
+    pub fn new(aabb: Aabb3d, origin: Vec3, direction: Direction3d, max: f32) -> Self {
+        Self::from_ray(aabb, Ray3d { origin, direction }, max)
+    }
+
+    /// Construct an [`AabbCast3d`] from an [`Aabb3d`], [`Ray3d`] and max distance.
+    pub fn from_ray(aabb: Aabb3d, ray: Ray3d, max: f32) -> Self {
+        Self {
+            ray: RayTest3d::from_ray(ray, max),
+            aabb,
+        }
+    }
+
+    /// Get the distance at which the [`Aabb3d`]s collide, if at all.
+    pub fn aabb_collision_at(&self, mut aabb: Aabb3d) -> Option<f32> {
+        aabb.min -= self.aabb.max;
+        aabb.max -= self.aabb.min;
+        self.ray.aabb_intersection_at(&aabb)
+    }
+}
+
+impl IntersectsVolume<Aabb3d> for AabbCast3d {
+    fn intersects(&self, volume: &Aabb3d) -> bool {
+        self.aabb_collision_at(*volume).is_some()
+    }
+}
+
+/// An intersection test that casts a [`BoundingSphere`] along a ray.
+#[derive(Debug)]
+pub struct BoundingSphereCast {
+    /// The ray along which to cast the bounding volume
+    pub ray: RayTest3d,
+    /// The sphere that is being cast
+    pub sphere: BoundingSphere,
+}
+
+impl BoundingSphereCast {
+    /// Construct a [`BoundingSphereCast`] from a [`BoundingSphere`], origin, [`Direction3d`] and max distance.
```

Comment:
Very much a nit: I was told in a review once that Bevy prefers the Oxford comma

(non-blocking lol)